libkrb5/src/context.rs

```diff
@@ -457,20 +455,36 @@ impl Krb5Context {
         [&tok_id, flags.as_slice(), filler, &seq_num].concat()
     }
 
+    fn verify_mic_token_header(header: &[u8], usage: Krb5KeyUsage, expected_seq_num: Option<i32>) -> Result<(), Krb5Error> {
+        let mut parse_mic_token_header = tuple::<_, _, nom::error::Error<&[u8]>, _>((be_u16, be_u8, take(5u8), be_u64));
+        let (_, (token_id, flags, filler, seq_num)) = parse_mic_token_header(header)?;
+
+        let expected_flags = Krb5Context::get_token_flags(usage);
+        if token_id != TOK_MIC_MSG || flags != expected_flags || filler != b"\xFF\xFF\xFF\xFF\xFF" {
+            return Err(Krb5Error::InvalidToken)
+        }
+
+        match expected_seq_num {
+            Some(expected_seq_num) if expected_seq_num != seq_num as i32 => return Err(Krb5Error::InvalidToken),
+            _ => Ok(())
+        }
+    }
+
     pub fn create_wrap_token_header(usage: Krb5KeyUsage, seq_num: i32, rrc: Option<u16>) -> Vec<u8> {
         let tok_id = TOK_WRAP_MSG.to_be_bytes();
         let flags = Krb5Context::get_token_flags(usage).to_be_bytes();
         let filler = b"\xFF";
         let ec: u16 = 0;
         let rrc: u16 = rrc.unwrap_or(0); /* rrc should be zero in the encrypted header */
-        let seq_num = seq_num as i64;
+        let seq_num = (seq_num as u64).to_be_bytes();
```

Comment:
#question: why do we need to cast to u64 for instead of i64?

Reply:
The sequence number is implemented as i32 in the lib, so I think i64 would be a better choice here.